scripts/ci/check_compliance.py

```diff
@@ -1446,44 +1446,39 @@ class Identity(ComplianceTest):
 
     def run(self):
         for shaidx in get_shas(COMMIT_RANGE):
-            commit = git("log", "--decorate=short", "--no-use-mailmap", "-n 1", shaidx)
-            signed = []
-            author = ""
-            sha = ""
-            parsed_addr = None
-            for line in commit.split("\n"):
-                match = re.search(r"^commit\s([^\s]*)", line)
-                if match:
-                    sha = match.group(1)
-                match = re.search(r"^Author:\s(.*)", line)
-                if match:
-                    author = match.group(1)
-                    parsed_addr = parseaddr(author)
-                match = re.search(r"signed-off-by:\s(.*)", line, re.IGNORECASE)
-                if match:
-                    signed.append(match.group(1))
-
-            error1 = f"{sha}: author email ({author}) needs to match one of " \
-                     f"the signed-off-by entries."
-            error2 = f"{sha}: author email ({author}) does not follow the " \
-                     f"syntax: First Last <email>."
-            error3 = f"{sha}: author email ({author}) must be a real email " \
-                     f"and cannot end in @users.noreply.github.com"
-            failure = None
-            if author not in signed:
-                failure = error1
-
-            if not parsed_addr or len(parsed_addr[0].split(" ")) < 2:
-                if not failure:
-
-                    failure = error2
+            delim = '||!!||'
+            fmt = delim.join(['%H', '%an', '%ae', '%b'])
+            commit = git("log", f'--format="{fmt}"', "--no-use-mailmap", "-n 1", shaidx)
+            (sha, auth_name, auth_email, body) = commit.split(delim)
+
+            match_signoff = re.search(r"signed-off-by:\s(.*)", body,
+                                      re.IGNORECASE)
+            detailed_match = re.search(r"signed-off-by:\s(.*) <(.*)>", body,
+                                       re.IGNORECASE)
+
+            failures = []
+
+            if auth_email.endswith("@users.noreply.github.com"):
+                failures.append(f"""{sha}: author email ({auth_email}) must be
+                                a real email and cannot end in
+                                @users.noreply.github.com""")
```

Comment:
nit: I think this and the others messes up the message by actually printing the indentation, can you do multiple single line strings instead?